src/app/application.cpp

```diff
@@ -790,8 +799,14 @@ try
         disconnect(m_desktopIntegration, &DesktopIntegration::activationRequested, this, &Application::createStartupProgressDialog);
         // we must not delete menu while it is used by DesktopIntegration
         auto *oldMenu = m_desktopIntegration->menu();
-        const MainWindow::State windowState = (!m_startupProgressDialog || (m_startupProgressDialog->windowState() & Qt::WindowMinimized))
-                ? MainWindow::Minimized : MainWindow::Normal;
+#ifndef Q_OS_MACOS
+        const WindowState windowState = !m_startupProgressDialog ? WindowState::Hidden
+                : (m_startupProgressDialog->windowState() & Qt::WindowMinimized) ? WindowState::Minimized
+                        : WindowState::Normal;
+#else
+        const WindowState windowState = (m_startupProgressDialog->windowState() & Qt::WindowMinimized)
```

Comment:
Null test `m_startupProgressDialog`?

Reply:
> Null test `m_startupProgressDialog`?

In case of macOS it is always created above.